bdai_ros2_wrappers/bdai_ros2_wrappers/time.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2024 Boston Dynamics AI Institute Inc.  All rights reserved.
+
+from datetime import datetime, timedelta
+from typing import Union
+
+from rclpy.duration import Duration
+from rclpy.time import Time
+
+
+def as_proper_time(time: Union[int, float, datetime, Time]) -> Time:
+    """Return `time` as a proper Time object.
+
+    Note that for scalar times the convention is that floating point times
+    are expressed in seconds since some clock epoch, and integral times are
+    expressed in nanoseconds since some clock epoch.
+    """
+    if isinstance(time, int):
+        return Time(nanoseconds=time)
+    if isinstance(time, float):
+        return Time(seconds=time)
+    if isinstance(time, datetime):
+        return Time(seconds=time.timestamp())
+    return time
+
+
+def as_proper_duration(duration: Union[int, float, timedelta, Duration]) -> Duration:
+    """Return `duration` as a proper Duration object.
+
+    Note that for scalar durations the convention is that floating point durations
```

Comment:
Do we want to add a proper docstring with input specification including units?

Reply:
Hmm, I'm not exactly sure what you mean by _input specification_.

Reply:
I meant, something like the following:

```suggestion
    """
    Convert various duration inputs into a standardized Duration object.

    Parameters:
    duration (Union[int, float, timedelta, Duration]): The duration to be converted.
        - If an int is provided, it is assumed to be the duration in seconds.
        - If a float is provided, it is assumed to be the duration in seconds.
        - If a timedelta is provided, it is converted to the appropriate Duration object.
        - If a Duration object is provided, it is returned as is.

    Returns:
    Duration: A standardized Duration object representing the input duration.

    Raises:
    ValueError: If the input duration is not of a supported type.

    Example:
    >>> as_proper_duration(3600)
    <Duration: 3600 seconds>
    >>> as_proper_duration(2.5)
    <Duration: 2.5 seconds>
    >>> as_proper_duration(timedelta(hours=1))
    <Duration: 3600 seconds>
    >>> as_proper_duration(Duration(3600))
    <Duration: 3600 seconds>
    """
```

Reply:
I meant, something like the following:

```
    """
    Convert various duration inputs into a standardized Duration object.

    Parameters:
    duration (Union[int, float, timedelta, Duration]): The duration to be converted.
        - If an int is provided, it is assumed to be the duration in seconds.
        - If a float is provided, it is assumed to be the duration in seconds.
        - If a timedelta is provided, it is converted to the appropriate Duration object.
        - If a Duration object is provided, it is returned as is.

    Returns:
    Duration: A standardized Duration object representing the input duration.

    Raises:
    ValueError: If the input duration is not of a supported type.

    Example:
    >>> as_proper_duration(3600)
    <Duration: 3600 seconds>
    >>> as_proper_duration(2.5)
    <Duration: 2.5 seconds>
    >>> as_proper_duration(timedelta(hours=1))
    <Duration: 3600 seconds>
    >>> as_proper_duration(Duration(3600))
    <Duration: 3600 seconds>
    """
```

Reply:
See https://github.com/bdaiinstitute/ros_utilities/pull/103/commits/c3262c8e1f3c5091d6420d5c1450352e90a5dfe4. I purposely opted out from duplicating type annotations in docstrings.